limits/write_card.py

```diff
@@ -0,0 +1,368 @@
+                        shapes_lines_toWrite.append("\n")
+                    else: 
+                        lnN_lines_toWrite.append(col1.format(proc+"_"+select+"_"+variable+"_"+regionSuffix[region]))
+                        lnN_lines_toWrite.append(cols.format("shape"))
+                        for lineproc in backgrounds: lnN_lines_toWrite.append(cols.format("-"))
+                        for lineproc in signals:     lnN_lines_toWrite.append(cols.format(systL[1] if lineproc == proc else "-"))
+                        lnN_lines_toWrite.append("\n")
+
+        ########################
+
+        file.write    ('imax *  number of channels\n')
+        file.write    ('jmax *  number of processes minus 1\n')
+        file.write    ('kmax *  number of nuisance parameters\n')
+        file.write    ('----------------------------------------------------------------------------------------------------------------------------------\n')
+        ## shapes
+
+        file.write    ('shapes * %s %s $PROCESS $PROCESS_$SYSTEMATIC\n'%(select, opt.filename))
+        file.write    ('----------------------------------------------------------------------------------------------------------------------------------\n')
+
+        file.write    ((col1+cols+'\n').format('bin', select)) ### blind for now
+        ## observation
+        file.write    ((col1+cols+'\n').format('observation', '-1')) ### blind for now
+
+        file.write    ('----------------------------------------------------------------------------------------------------------------------------------\n')
+        ## processes 
+        file.write    ('# list the expected events for signal and all backgrounds in that bin\n')
+        file.write    ('# the second process line must have a positive number for backgrounds, and 0 or neg for signal\n')
+        file.write    (col1.format('bin'))
+        for i in range(0,len(backgrounds)+len(signals)): 
+            file.write(cols.format(select))
+        file.write    ("\n")
+        file.write    (col1.format('process'))
+        for proc in backgrounds: 
+            file.write(cols.format(proc))
+        for proc in signals: 
+            file.write(cols.format(proc))
+        file.write    ("\n")
+        file.write    (col1.format("process"))
+        for i in range(1,len(backgrounds)+1): 
+            file.write(cols.format(i))
+        for i in range(0,len(signals)): 
+            file.write(cols.format(str(-int(i))))
+        file.write    ("\n")
+        file.write    (col1.format("rate"))
+        for proc in range(len(backgrounds)+len(signals)):
+            file.write(ratecols.format(rates[proc]))
+        file.write    ("\n")
+        file.write    ('----------------------------------------------------------------------------------------------------------------------------------\n')
+
+        if region ==0:        
```

Comment:
I have some dummy problem here that I don't manage to solve: from here on, ```file.write(whatever_line)``` has no effect, although I do enter in the loop. I don't understand if it's a matter of i/o streams that get lost or of trivial stupidity...